### ISSUE TITLE
Always on Top for any window?

### ISSUE THREAD

**Issue by DaHO223** [Kolcha/DigitalClock5#12]
Hello! Will the plugin "Always on Top for any window" be available for the new version again? Thanks, Dave

**Comment by Kolcha** [comment]
hello!

I am surprised that someone uses it all... it was implemented "just for fun" very long time ago because I missed "always on top" option for any window on Windows that is available in many Linux environments "out-of-the-box".

it is easy to add this plugin, as it basically does nothing - it just loads another library with actual "hook". but I considered not adding it just because of this "eternal" (since ~2015) issue with "stay on top"... nowadays it is just useless... so, I don't want to add it, to be honest, just to avoid another bunch of complaints about non-working "stay on top"...

yes, it may work +/- reliably (for example in my environment, as I just don't have any apps that potentially may lead to this unstable behavior), the only "mystery" is in Windows itself.

if you don't experience the issues with "stay on top" option, this very likely will work for you too, but beware - Windows can maintain only one "on top" window (at least it was so, can't find any confirmation now), so clock window itself may lose "on top" property as result of this plugin. also, due to set of hacks in clock to get this "stay on top" feature work at least somehow reliably, effect from this plugin also will be very unstable. so, the value from it is very questionable...

to summarize: at this moment I have no plans to add it, but will see.

**Comment by Kolcha** [comment]
closing as "won't fix"

**Comment by Kolcha** [comment]
problem with "stay on top" is "number one" problem since ~2015 and the underlying reason is still totally unclear to me but looks like I have no control over it. I added many hacks through the years (in the previous Digital Clock 4, then these hack were migrated to 5) to mitigate the problem, but usually hacks stopped working after some time (I think after some Windows update). for more details see [my FAQ](https://github.com/Kolcha/DigitalClock5/wiki/Frequently-asked-questions), you will find many my findings there (info from "link 1" is bit outdated - app restart is not required, I implemented hack). it works in many cases, but not always, issue appears from system to system, so that this issue is marked as "won't fix".

> The window needs the extended window style called WS_EX_TOPMOST

I know this, and there is [the code](https://github.com/Kolcha/DigitalClock5/blob/master/app/platform/win/clock_native_window.cpp#L95) (hack) which check it periodically and sets if it was "lost".

> as a developer, I have a Windows application that does stay atop other window

if you are a Windows developer, likely you know more than me and maybe can confirm my findings. I barely remember (I worked with Win API only around 2010) that Windows can maintain one "always on top" window, and drops the mentioned flag from other windows. likely some other window "steals" this flag very aggressive (runs few ms timer and sets it again and again), this is the reason why "stay on top" doesn't work for you. you may try to find what app does this.

if would be nice if you share your findings.

**Comment by mcsilver** [comment]
Ah, I see from your code that you did try to ensure that the window has WS_EX_TOPMOST. My bad.

Also, for some reason, the window now has WS_EX_TOPMOST for me and it is behaving correctly--I'm not sure what changed.

I read that in Qt, a window can sometimes get recreated, so I will monitor my running instance of the app. If it loses its topmost behaviour, then I will check for a changed window handle.
